content/extractor.js

```diff
@@ -16,36 +16,122 @@
 
 // Given DOM element returns array of possible id-links to resolve.
 
+import { flatten } from 'lodash';
+
 export const TAGS_BLACKLIST = [ 'script' ];
 
-export function extractPossibleMatches ($dom) {
-  const tag = $dom.tagName.toLowerCase();
+const EMAIL_PATTERN = /([^\s@]+@[^\s@]+\.[a-z]+)/i;
```

Comment:
IMHO it's fine, we can even list all top level domains here and afair all of them are alpha-only.